File: app/src/main/java/org/odddev/fantlab/core/network/NetworkModule.java
package org.odddev.fantlab.core.network;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author kenrube
 * @since 15.09.16
 */

@Module
public class NetworkModule {

    @Provides
    @Singleton
    IServerApi provideServerApi() {
        return ServerApiBuilder.createApi();
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/network/NetworkModule.kt
package org.odddev.fantlab.core.network

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author kenrube
 * *
 * @since 15.09.16
 */

@Module
class NetworkModule {

	@Provides
	@Singleton
	fun provideServerApi() = ServerApiBuilder.createApi()
}
